frontend/src/app/projects/models/model-description/model-description.component.html

```diff
@@ -51,9 +51,19 @@
         <a mat-button [routerLink]="'../../..'"
           ><mat-icon>arrow_back</mat-icon>Back</a
         >
-        <button mat-flat-button color="primary" type="submit">
-          Update model
-        </button>
+        <div>
+          <button
+            mat-flat-button
+            color="warn"
+            (click)="deleteModel()"
+            *ngIf="canDelete"
+          >
```

Comment:
```suggestion
          <button
            mat-flat-button
            color="warn"
            (click)="deleteModel()"
            *ngIf="canDelete"
            type="button"
          >
```
The default type of the button here is `submit` and therefore the `onSubmit()` function is called when the cancel button is clicked.

In addition, the delete button is not visible if there are still git/t4c models. I think it would be better to use a solution here that shows that you are normally able to delete a model, but that this is currently not possible due to existing git/t4c models. Here are some solutions I could think of:
- disable the button and add a tooltip with something like *Please remove all Git and T4C models first before deleting this model*
- disable the button and add an information text above it that prompts the user to remove the models first.
- do the same thing Moritz did with deleting tool versions (and other things): Always leave the button enabled, but if you click on it and there are still dependencies, show them in the error message

Reply:
> I do not like this, since it clutters the interface with things you cannot do anyway. 

Isn't this an argument for the tooltip? The tooltip only shows the information if you hover over the `Delete` button. Without the tooltip, I see the information about the deletion - even if I just want to update the model, which may be confusing. 
You want to update the model and it shows: `You can not delete the model because of linked Git / T4C models`. 

The tooltip message only appears when you intend to click the Delete button.